components/tikv_util/src/metrics/memory_linux.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! This module is a subset of rust-prometheus's process collector, without the fd collector
+//! to avoid memory fragmentation issues when open fd is large.
+
+use std::io::{Error, ErrorKind, Result};
+
+use libc;
+use procinfo::pid as pid_info;
+
+use prometheus::core::{Collector, Desc};
+use prometheus::{proto, Gauge, Opts};
+
+/// Monitors threads of the current process.
+pub fn monitor_memory() -> Result<()> {
+    let pid = unsafe { libc::getpid() };
+    let tc = MemoryCollector::new(pid);
+    prometheus::register(Box::new(tc)).map_err(|e| Error::new(ErrorKind::Other, e.to_string()))
+}
+
+/// A collector to collect memory metrics.
+pub struct MemoryCollector {
+    pid: libc::pid_t,
+    descs: Vec<Desc>,
+    vsize: Gauge,
+    rss: Gauge,
+}
+
+impl MemoryCollector {
+    pub fn new(pid: libc::pid_t) -> Self {
+        let mut descs = Vec::new();
+
+        let vsize = Gauge::with_opts(Opts::new(
```

Comment:
Can it be solved by node exporter?

Reply:
Node exporter exports the memory usage of node exporter itself and looks like there is [no config](https://prometheus.io/docs/guides/node-exporter/) for the node exporter.